### DISCUSSION TITLE
How to display Math on Github-Pages?

### DISCUSSION THREAD

**Discussion by ghost** [community/community#22525] [Other Feature Feedback, Questions, & Ideas]
<p>Hi, Is there a way to display Math from a markdown file on Github pages? (I know that it’s not possible for Github, but I would like to know if there’s an option to make it work for GH-Pages).</p>
<p>I’ve done a lot of <code>tab-research</code> but couldn’t come up with anything. Almost all of the content is geared towards the blogging system of Jekyll (or using it in HTML). However, I don’t want to use the blog feature of Jekyll, I just want to display simple math in my README.md (and other relative markdown files) on GH-Pages.</p>
<p>A link to the relevant topic would be much appreciated.</p>
<p>Thanks!</p>

**Comment by lee-dohm** [comment]
<p>If you simply allow GitHub Pages to render your Markdown files into HTML without customizing the rendering yourself via something like Jekyll or another static site generator, then no. The default rendering of Markdown to HTML via GitHub Pages doesn’t include anything like MathJax for displaying formatted math formulae.</p>
<p>If you use Jekyll (or another static site generator), then you could change the default template to include MathJax like I’ve done on my GitHub Pages website:</p>
<aside class="onebox githubblob">
  <header class="source">
      <a href="https://github.com/lee-dohm/lee-dohm.github.io/blob/77260089817da592b570b64a33f5642dca60a9b9/_layouts/base.html#L20-L23" target="_blank">github.com</a>
  </header>
  <article class="onebox-body">
    <h4><a href="https://github.com/lee-dohm/lee-dohm.github.io/blob/77260089817da592b570b64a33f5642dca60a9b9/_layouts/base.html#L20-L23" target="_blank">lee-dohm/lee-dohm.github.io/blob/77260089817da592b570b64a33f5642dca60a9b9/_layouts/base.html#L20-L23</a></h4>
<pre class="onebox"><code class="lang-html"><ol class="start lines" start="20" style="counter-reset: li-counter 19 ;">
<li>&lt;!-- MathJax --&gt;</li>
<li>&lt;script type="text/javascript"</li>
<li>  src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.3/MathJax.js?config=TeX-AMS-MML_HTMLorMML"&gt;</li>
<li>&lt;/script&gt;</li>
</ol></code></pre>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>I hope that helps!</p>

**Comment by ghost** [comment]
<p>I have no issues using Jekyll to generate my site (and then pushing the HTML). However, when I tried to use Jekyll, all I can see is the blogging feature.</p>
<p>To be more specific, say, I have a <code>README.md</code> file in my repo (with a relative link to <code>foo/bar.md</code> . (Both the markdown files contain math).</p>
<p>Now, I want to use Jekyll to convert these markdown files to HTML (with math included). I don’t want to use any other feature of Jekyll (more specifically, I don’t need a <code>posts</code> folder).</p>
<p>So, how would I go about implementing this?</p>
<p>I understand it’s not a domain of Github, but thanks in advance.</p>
<p>(Or is it not possible to convert a set of already existing markdown files on Github to HTML via Jekyll)?</p>

**Comment by lee-dohm** [comment]
<p>You can use Jekyll without the blog functionality. If you take a look at the repository for my website, you’ll notice that I have a mixture of <a href="https://github.com/lee-dohm/lee-dohm.github.io/blob/77260089817da592b570b64a33f5642dca60a9b9/_posts/2010-08-08-files-are-an-implementation-detail.md">blog</a> and <a href="https://github.com/lee-dohm/lee-dohm.github.io/blob/77260089817da592b570b64a33f5642dca60a9b9/about.md">non-blog</a> content in there. The blog functionality is there to optimize for the most common use of static site generators, blog-like structures. If you want to have just a bunch of pages that you manually control the structure of, you can do that by linking to where they’re going to be generated into position like I do here:</p>
<aside class="onebox githubblob">
  <header class="source">
      <a href="https://github.com/lee-dohm/lee-dohm.github.io/blob/77260089817da592b570b64a33f5642dca60a9b9/_layouts/base.html#L57-L61" target="_blank">github.com</a>
  </header>
  <article class="onebox-body">
    <h4><a href="https://github.com/lee-dohm/lee-dohm.github.io/blob/77260089817da592b570b64a33f5642dca60a9b9/_layouts/base.html#L57-L61" target="_blank">lee-dohm/lee-dohm.github.io/blob/77260089817da592b570b64a33f5642dca60a9b9/_layouts/base.html#L57-L61</a></h4>
<pre class="onebox"><code class="lang-html"><ol class="start lines" start="57" style="counter-reset: li-counter 56 ;">
<li>&lt;li&gt;&lt;a href="/about"&gt;About Lee&lt;/a&gt;&lt;/li&gt;</li>
<li>&lt;li&gt;&lt;a href="/this-blog"&gt;About This Blog&lt;/a&gt;&lt;/li&gt;</li>
<li>&lt;li&gt;&lt;a href="/appearances"&gt;Appearances&lt;/a&gt;&lt;/li&gt;</li>
<li>&lt;li&gt;&lt;a href="/colophon"&gt;Colophon&lt;/a&gt;&lt;/li&gt;</li>
<li>&lt;li&gt;&lt;a href="/disclaimer"&gt;Disclaimers&lt;/a&gt;&lt;/li&gt;</li>
</ol></code></pre>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by ghost** [comment]
<p>Thanks for the insights. I’ll take a look at your repo and (hopefully) setup my own. Have a good day!</p>

**Comment by ghost** [comment]
<p>Also, slightly off topic, but do you know a workaround for jumping from GH-Pages to Github? The rules for redirection are fairly simple, (as you just need to add <code>blob/master</code> and do a small change in the url (which is just 1 or 2 lines of code using <code>regex</code>). I was surprised there wasn’t any plugins for this yet. My question is : Can I somehow tweak the source code to include a button that lets me jump to the file location on Github?</p>

**Comment by lee-dohm** [comment, marked as answer]
<aside class="quote no-group" data-username="Just-A-Visitor" data-post="6" data-topic="116439">
<div class="title">
<div class="quote-controls"></div>
 Just-A-Visitor:</div>
<blockquote>
<p>Can I somehow tweak the source code to include a button that lets me jump to the file location on Github?</p>
</blockquote>
</aside>
<p>You can control all kinds of things like this when you use Jekyll (or another static site generator). I know that my friend and coworker <a class="mention" href="/benbalter">@benbalter</a> has some elements that I have yet to <s>steal</s> lovingly copy from <a href="https://ben.balter.com">his website</a> such as a link to view <a href="https://github.com/benbalter/retlab/blob/116130c0d70d8f6193adf75d3095aa488075351b/_layouts/post.html#L13">a page’s revision history</a> or <a href="https://github.com/benbalter/retlab/blob/116130c0d70d8f6193adf75d3095aa488075351b/_includes/mini-bio.html#L11-L14">submit a pull request to improve the page</a> with the help of the <a href="https://github.com/jekyll/github-metadata">jekyll-github-metadata plugin</a>:</p>
<p><img src="https://user-images.githubusercontent.com/1038121/181071619-9d267bcc-74c1-42dc-b9a8-a34bff4eb529.png" alt="Screen Shot 2020-06-02 at 12.59.30 PM" data-base62-sha1="pQjOhC3i7MtiAGshLzwBnxmEWjr" width="626" height="128"></p>
<p><img src="https://user-images.githubusercontent.com/1038121/181071620-248e84ea-0ceb-4790-b0d3-775b757ab62b.png" alt="Screen Shot 2020-06-02 at 12.59.39 PM" data-base62-sha1="1siJj4kizeKoq6B0DpXqpiRWZty" width="334" height="262"></p>
<p>Basically, if your question is some derivative of “can I customize pages on GitHub Pages to … ?”, the answer is invariably going to be “with a static site generator such as Jekyll, yes” :grinning:</p>
<p>But the reason why you probably haven’t seen a plugin specifically for doing exactly what you described is that there are many reasons for wanting to link back to GitHub from a page. The jekyll-github-metadata plugin allows you to implement any of them.</p>

**Comment by ghost** [comment]
<p>Right, <code>Improve on Github</code> as mentioned <a href="https://github.com/jekyll/github-metadata/blob/c0d0fa8057c02bb5de23af90b4b8c27e529350d7/docs/edit-on-github-link.md" rel="nofollow noopener">here</a> was exactly what I was looking for.</p>
<p>I think I can take it from here. Thanks for the help!</p>

**Comment by nschloe** [comment]
<p>A while ago I created <a href="https://github.com/nschloe/xhub" rel="noopener nofollow ugc">xhub</a>, a browser extension that allows you to use math in github pages.</p>
<p>Cons:</p>
<ul>
<li>You have to install the extension.</li>
</ul>
<p>Pros:</p>
<ul>
<li>No need to set up any workflow.</li>
<li>Just edit your markdown as usual and use<pre><code class="lang-markdown">Display math:
```math
e^{i\pi} + 1 = 0
```
and inline math $`a^2 + b^2 = c^2`$.
</code></pre>
</li>
<li>Works well on light and dark background.</li>
<li>You can even copy-and-paste the math!</li>
</ul>
<p>Perhaps worth checking out.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/e/d/edc74f6e525efcd72f82d76c882a5005e9aae47c.png" data-download-href="/uploads/short-url/xVubcYH57JpbTH2DfCa2YyUkUB6.png?dl=1" title="screenshot" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/181628/181071623-9458d2a3-971d-4423-84c2-318ff6869166.png" alt="screenshot" data-base62-sha1="xVubcYH57JpbTH2DfCa2YyUkUB6" width="690" height="233" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/e/d/edc74f6e525efcd72f82d76c882a5005e9aae47c_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">screenshot</span><span class="informations">1178×398 21.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by sahidyk** [comment]
<p>Very interesting and useful. The extension can also be used in Edge.</p>
<p>However, the pages can not display the mathematics properly when the pages are built into web in Github. It only works on the original Github Markdown files.</p>

**Comment by jumpjack** [comment]
What does it mean using a chrome extension to add math to github pages? Will all visitors be able to properly see my formulas?!?
